xdsl/dialects/riscv_func.py

```diff
@@ -48,9 +51,102 @@ def verify_(self):
             )
 
 
+@irdl_op_definition
+class CallOp(IRDLOperation):
+    """RISC-V function call operation"""
+
+    name = "riscv_func.call"
+    args: Annotated[VarOperand, riscv.RegisterType]
+    func_name: OpAttr[StringAttr]
+    result: Annotated[OptOpResult, riscv.RegisterType]
+
+    def __init__(
+        self,
+        func_name: StringAttr,
+        args: list[Operation | SSAValue],
+        has_result: bool = True,
+        comment: StringAttr | None = None,
+    ):
+        super().__init__(
+            operands=[args],
+            result_types=[riscv.RegisterType(riscv.Register()) if has_result else None],
+            attributes={
+                "func_name": func_name,
+                "comment": comment,
+            },
+        )
+
+
+@irdl_op_definition
+class FuncOp(IRDLOperation):
+    """RISC-V function definition operation"""
+
+    name = "riscv_func.func"
+    args: Annotated[VarOperand, riscv.RegisterType]
+    func_name: OpAttr[StringAttr]
+    func_body: SingleBlockRegion
+
+    def __init__(self, name: str, region: Region):
+        attributes: dict[str, Attribute] = {"func_name": StringAttr(name)}
+
+        super().__init__(attributes=attributes, regions=[region])
+
+    def verify_(self):
+        # Check that the returned value matches the type of the function
+        if len(self.func_body.blocks) != 1:
+            raise VerifyException("Expected FuncOp to contain one block")
+
+        block = self.func_body.blocks[0]
+
+        if not len(block.ops):
```

Comment:
I think we should stick to explicit comparisons here, so `len(block.ops) == 0`. I feel that that is much easier to read and more in line with out codebase.